pkg/platform/runtime/setup/buildlog/messages.go

```diff
@@ -0,0 +1,164 @@
+package buildlog
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+)
+
+// MessageEnum enumerates the build events that can be expected on the buildlog websocket connection
+type MessageEnum int
+
+const (
+	BuildStarted MessageEnum = iota
+	BuildSucceeded
+	BuildFailed
+	ArtifactStarted
+	ArtifactSucceeded
+	ArtifactFailed
+	ArtifactProgress
+	Heartbeat
+	UnknownMessage
```

Comment:
nit: Similar to this, I suggest using 0 for something like Unset. That way, if a struct has a MessageEnum field, the zero value makes it clear that it was not set.